docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,585 @@
+[[runtime]]
+=== Runtime
+Typically, you must index fields to {es} before they can be retrieved,
+aggregated, or searched. With _runtime fields_ beta[], you can explicitly
+define a field in the mapping and access it at search time without indexing
+your data during ingest time.
+
+Because runtime fields aren't indexed, you can more quickly ingest raw data
+into the Elastic Stack and access it immediately. Runtime fields are accessible
+from the search API like any other field. The <<search-field-caps,field capabilities API>> returns runtime fields as both searchable and aggregatable.
+
+By dynamically evaluating runtime fields at search time, you can optimize disk
+space by choosing which fields to index. If you later decide that a runtime
+field is increasingly used for filtering or aggregation, you can add the field
+to `_source` to gain faster search speed.
+
+Alternatively, you can specify a `runtime_mappings` section in a search request
+to use runtime fields that exist only as part of the query. This
+flexibility allows you to create or override field values in `_source` for the
+duration of the query without modifying the field itself.
+
+Runtime fields help to alleviate several common issues when using {es}:
+
+* Reindexing your data between development iterations is slow and can make
```

Comment:
* Understanding you want to add a field after the documents are already indexed
* Wanting to immediately start working on a new datastream, without fully understanding the data in it, and only while working on it (using runtime fields based on _source), understand what's the schema that will be most helpful and gradually adding it
* Shadowing an indexed field with a runtime field to fix a mistake after the documents are indexed
* Defining fields that are only relevant for a particular context, e.g. a certain visualization, as ephemeral fields that are defined as part of the query, without influencing everyone's schema

Reply:
Thanks for these @giladgal! I integrated these points into the [use cases](https://elasticsearch_62653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime.html#runtime-use-cases).